DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -73,5 +76,30 @@ public List<CentreUserDetails> GetUserCentres(AdminUser? adminUser, List<Delegat
 
             return availableCentres.OrderByDescending(ac => ac.IsAdmin).ThenBy(ac => ac.CentreName).ToList();
         }
+
+        public bool TryUpdateUserAccountDetails(string password, string signedInEmail, string firstName, string surname, string email)
+        {
+            var (adminUsers, delegateUsers) = GetUsersByEmailAddress(signedInEmail);
+
+            var (verifiedAdminUser, verifiedDelegateUsers) =
+                loginService.VerifyUsers(password, adminUsers.SingleOrDefault(), delegateUsers);
+
+            if (verifiedAdminUser == null && verifiedDelegateUsers.Count == 0)
+            {
+                return false;
+            }
+
+            if (verifiedAdminUser != null)
+            {
+                userDataService.UpdateAdminUser(firstName, surname, email, verifiedAdminUser.Id);
+            }
+
+            foreach (var verifiedDelegateUser in verifiedDelegateUsers)
+            {
+                userDataService.UpdateDelegateUser(firstName, surname, email, verifiedDelegateUser.Id);
```

Comment:
Would it be worth having a single query based on a list of ids? Then rollbacks will be handled by default too if any update fails.